### PR TITLE
Rename abort JS helper function

### PR DESCRIPTION
**Pull request by sbc100**
Also, remove the comment which I believe is not longer relevant with this change.

### CODE REVIEW

File: system/lib/libc/musl/src/exit/abort.c
Comment (kripken) [inline system/lib/libc/musl/src/exit/abort.c:17]: What's the advantage of including this `.c` file if it just forwards out to JS? Implementing it in JS directly seems simpler.

File: src/library.js
Comment (kripken) [inline src/library.js:405]: I think the TODO is still relevant, in that in theory we could imagine not going through this "trampoline" here. But closure will inline it anyhow so I'm not sure if it matters.

Comment (sbc100) [reply src/library.js:405]: My current thinking is that trampiline is useful since it injects the "native code aborted" message before calling the JS abort function.     So I don't think should be trying to remove it anymore.

File: system/lib/libc/musl/src/exit/abort.c
Comment (sbc100) [reply system/lib/libc/musl/src/exit/abort.c:17]: Sorry I should have mentioned, this was needed as part of an experiment I'm doing to remove the underscore name mangling.   One place where that is an issue is where the new library function name `abort` vs `_abort` clashed with some existing JS symbol, in this case the JS abort utility defined in the preamble.

Avoiding these clashing names seems like a good thing to avoid confusion anyway.. since if you read the current JS abort library function it looks like its infinite recursion, when in fact its calling from the C-facing abort to the JS-flavored abort (that later takes a string argument).

**Comment by sbc100** [comment]
I cant put this on hold for now until that experiment is closer to landing.

File: system/lib/libc/musl/src/exit/abort.c
Comment (kripken) [reply system/lib/libc/musl/src/exit/abort.c:17]: I see. Yeah, as a workaround for that problem this sounds reasonable (with a comment in the source maybe saying that is why it is needed).

**Comment by kripken** [comment]
(can or can't in the last comment?)

**Comment by sbc100** [comment]
> (can or can't in the last comment?)

Sorry, I can/will put this on hold.

**Comment by sbc100** [comment]
Actually I think this change is good to go now.

I've also improved the standalone version of abort to trap rather than exit(1).

File: src/library.js
Comment (kripken) [inline src/library.js:401]:
```suggestion
  // Use to implement the native `abort` symbol.  Note that we use the
```

File: src/library.js
Comment (kripken) [inline src/library.js:402]:
```suggestion
  // JavaScript `abort` helper in order to implement this function,
```

File: src/library.js
Comment (kripken) [inline src/library.js:403]:
```suggestion
  // but we use a distinct name here to avoid confusing the two.
```

File: system/lib/libc/musl/src/exit/abort.c
Comment (kripken) [inline system/lib/libc/musl/src/exit/abort.c:16]:
```suggestion
	/* In emscripten we call out JS to perform the actual abort where it can
```

File: system/lib/libc/musl/src/exit/abort.c
Comment (kripken) [inline system/lib/libc/musl/src/exit/abort.c:17]:
```suggestion
	 * produce a nice error.
```

File: system/lib/libc/musl/src/exit/abort.c
Comment (kripken) [inline system/lib/libc/musl/src/exit/abort.c:18]:
```suggestion
	 * Note that the JS library function not called `abort` to avoid conflicting with
```

File: system/lib/standalone/standalone.c
Comment (kripken) [inline system/lib/standalone/standalone.c:38]: Why did this change? And if it is correct, why is there anything after it?

Comment (sbc100) [reply system/lib/standalone/standalone.c:38]: Yes, I think this is correct.

I'm mostly copying from musl where it looks like the `_Exit()` call is there as double check: https://github.com/emscripten-core/emscripten/blob/d9aed286296e04b80f05b6f827d7434d1a3e115f/system/lib/libc/musl/src/exit/abort.c#L26-L29|

I will duplicate the comment here so it clear.

File: system/lib/libc/musl/src/exit/abort.c
Comment (dschuff) [inline system/lib/libc/musl/src/exit/abort.c:18]:
```suggestion
	 * Note that the JS library function is not called `abort` to avoid conflict with
```

File: system/lib/libc/musl/src/exit/abort.c
Comment (dschuff) [reply system/lib/libc/musl/src/exit/abort.c:22]: Can you also mention this in the commit description?

**Review by dschuff — APPROVED** [review]

**Pull request merged**
